### PR TITLE
Feature/filacms 17

### PR DESCRIPTION
**Pull request by jeremy-portable**

### CODE REVIEW

**Comment by jeremy-portable** [comment]
image

File: database/migrations/2024_03_20_000000_modify_contents_data_type.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            $table->json('contents')->change();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('pages');
```
Comment (kyoungportable) [inline database/migrations/2024_03_20_000000_modify_contents_data_type.php:23]: This isn't the reverse of the `up` operation.  Please correct it to just revert the change to the 'contents' column

File: src/Filament/Resources/AbstractContentResource.php [hunk 2]
```diff
@@ -146,15 +147,17 @@ public static function tiptapEditor($name = 'contents'): TiptapEditor
...
+            ->output(TiptapOutput::Json);
     }
 
     public static function table(Table $table): Table
     {
         return $table
             ->columns([
                 TextColumn::make('title')
-                    ->description(fn (Page $page): string => substr($page->contents, 0, 50) . '...')
+                    // ->description(fn (Page $page): string => substr($page->contents, 0, 50) . '...')
```
Comment (kyoungportable) [inline src/Filament/Resources/AbstractContentResource.php:160]: Just remove the line, no need to comment it out :)

File: src/Models/AbstractContentModel.php [hunk 3]
```diff
@@ -42,15 +43,27 @@ abstract class AbstractContentModel extends Model
...
 
+    public function shortDescription($length = 50, $omission = '...'): string
+    {
+        foreach ($this->contents['content'] as $key => $value) {
+            if ($value['type'] === 'paragraph') {
+                return Str::of($value['content'][0]['text'])->take($length) . $omission;
+            }
+        }
+
+        return '...';
```
Comment (kyoungportable) [inline src/Models/AbstractContentModel.php:64]: Should return `$omission` here?

Also, `shortDescription` should be a proper attribute

**Review by kyoungportable — CHANGES_REQUESTED** [review]
Please see comments.

Also, please add/adjust tests for the new functionality

File: database/migrations/2024_03_20_000000_modify_contents_data_type.php (hunk 1 above)
Comment (jeremy-portable) [reply database/migrations/2024_03_20_000000_modify_contents_data_type.php:23]: Yep sorry that shouldn't be there

**Review by kyoungportable — APPROVED** [review]
